agent/src/main.rs

```diff
@@ -187,7 +187,12 @@ async fn main() -> Result<()> {
         Err(_) => 8181,
     };
 
-    let metadata = PeerMetadata::new(format!("{}:{}", host, port), roles, None);
+    let metadata = PeerMetadata::new(
+        Uuid::new_v4().to_string(),
+        format!("http://{}:{}", host, port),
```

Comment:
Not a thing for this PR, but it would be ideal to start passing around `SocketAddr`s rather than `host` and `port`, because the way you stringify host is different for IPv4 and IPv6 (which needs `[brackets]` around the IP).